### DISCUSSION TITLE
RabbitMQ bus based on amqp streams and testing

### DISCUSSION THREAD

**Discussion by sarmis** [dotnetcore/EasyCaching#451] [Ideas]
Hello all,

I want to contribute a stream-based, RabbitMQ bus (which should be more suitable for systems with a significant amount of nodes). I have written some integration tests that check the behavior of the bus (and the hybrid provider), but the need rabbit mq to run. Can we accommodate that in the testing infrastructure? How should I proceed with the Pull Request regarding the integration tests?

**Comment by catcherwong** [comment]
> Can we accommodate that in the testing infrastructure? 

Yes.

> How should I proceed with the Pull Request regarding the integration tests?

Following this one https://docs.github.com/en/actions/using-containerized-services/about-service-containers

Adding a new github workflow for bus and hybrid.

**Reply by sarmis** [reply]
Thank you, I will try to follow the guide
